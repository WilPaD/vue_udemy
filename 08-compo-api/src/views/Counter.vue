<template>
    <h1>Counter Vue</h1>
    <h2>{{ counter }}</h2>

    <button @click="increase"> +1 </button>
    <button @click="decrease"> -1 </button>
</template>
<script>
import { ref } from 'vue';
import useCounter from '@/composables/useCounter'

export default {
    name: 'Counter',
    props:{},
    emits:{},
    setup(){
        /**
        const counter = ref(5)
         * Lo ideal es que las funciones se creen dentro del setup()
         * si se necesia de la referencia a la misma, de lo contrario
         * puede hacerse a cómo se dejó en el return pues sólo hace 
         * un incremento o decremento
        const increase = () => {
            counter.value++
        }
        const decrease = () => {
            counter.value--
        }
         //Aquí va todo lo que vamos a usar en el template
        return{
            //Objetos reactivos
            counter,

            //Métodos
            increase: () => counter.value++,
            decrease: () => counter.value--,
        }
        */

        //Aqui usamos el composable, el cual posee toda la lógica de lo que estamos haciendo
        const { counter, increase, decrease } = useCounter()

        return {
            counter,
            increase,
            decrease
        }
    }
}
</script>
<style>
    
</style>