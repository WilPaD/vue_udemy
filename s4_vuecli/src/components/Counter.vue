<template>
    <h2>Counter</h2>
    <p> 12 </p>
</template>

<script>
export default {
    name: 'Contador'
    
}
</script>

<style>

</style>