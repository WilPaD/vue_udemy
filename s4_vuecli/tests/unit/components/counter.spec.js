import {shallowMount, mount} from '@vue/test-utils'
import Counter from '@/components/Counter'


describe('Componente Counter', () => {
    
    test('Debe hacer match con el snapshot', () => {
        
        const wrapper = shallowMount( Counter )
        
        
        
        expect( wrapper.html() ).toMatchSnapshot()
    })

})