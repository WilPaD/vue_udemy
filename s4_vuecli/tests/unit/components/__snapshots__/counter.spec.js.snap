// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Componente Counter Debe hacer match con el snapshot 1`] = `
<h2>Counter</h2>
<p>10<sup>2</sup> = 100</p>
<div><button> +1 </button><button> -1 </button></div>
`;
